astropy/cosmology/units.py

```diff
@@ -150,23 +238,52 @@ def with_redshift(cosmology=None, *,
     ----------
     cosmology : `~astropy.cosmology.Cosmology`, str, or None, optional
         A cosmology realization or built-in cosmology's name (e.g. 'Planck18').
-        If None, will use the default cosmology
+        If `None`, will use the default cosmology
         (controlled by :class:`~astropy.cosmology.default_cosmology`).
 
-    Tcmb : bool (optional, keyword-only)
-        Whether to create a CMB temperature <-> redshift equivalency, using
-        ``Cosmology.Tcmb``. Default is `True`.
+    distance : {'comoving', 'lookback', 'luminosity'} or None (optional, keyword-only)
+        The distance type or `None`. Default is 'comoving'.
```

Comment:
Maybe "The type of distance equivalency to create or `None`...."

Reply:
👍